lib/rex/logging/log_dispatcher.rb

```diff
@@ -134,8 +134,39 @@ def dlog(msg, src = 'core', level = 0)
   $dispatcher.log(LOG_DEBUG, src, level, msg)
 end
 
-def elog(msg, src = 'core', level = 0)
-  $dispatcher.log(LOG_ERROR, src, level, msg)
+# Logs errors in a standard format for each Log Level.
+#
+# @param msg [String] Contains message from the developer explaining why an error was encountered. Log Levels 0-3.
+#
+# @param src [String] USed to indicate where the error is originating from. Most commonly set to 'core' to ensure logs
+# are place in 'framework.log'.
+#
+# @param level [Integer] Sets the level an error should be logged at. Descriptions of each log level can be found in
+# lib/rex/logging.rb. Log Level will be changeD automatically to 0 if no +error+ is included.
+#
+# @param error [Exception] Exception of an error that needs to be logged. Mandatory in Log Levels 1-2. Optional in Log Level 3.
+#
+# (Eg Loop Iterations, Variables, Function Calls).
+#
+# @return [NilClass].
+def elog(msg, src = 'core', error:nil)
+  if error.nil?
+    $dispatcher.log(LOG_ERROR, src, get_log_level(src), msg)
+    return
+  else
+
+    log_level = get_log_level(src)
+
+    unless log_level
+      log_level = LEV_3
+    end
+
+    error_details = "#{error.class} #{error.message}"
+    if get_log_level(src) >= LEV_1
+      error_details << "\nCall stack:\n#{error.backtrace.join("\n")}"
+    end
+    $dispatcher.log(LOG_ERROR, src, get_log_level(src), "#{msg} - #{error_details}")
```

Comment:
This looks like its passing down the current global log level, rather than the log level associated with this specific log message 👀

Reply:
Clarified after discussion